Makefile

```diff
@@ -561,6 +561,21 @@ nuke: clean clean-docker ## Clean plus removes persistant server data.
 setup-mac: ## Adds macOS hosts entries for Docker.
 	echo $$(boot2docker ip 2> /dev/null) dockerhost | sudo tee -a /etc/hosts
 
+update-dependencies: ## Uses go get -u to update all the dependencies while holding back any that require it. 
+	@echo Updating Dependencies
+
+	# Update all dependencies (does not update across major versions)
+	go get -u
```

Comment:
This is also one of the thing that you just "copied over" and I want to revisit this concept. Please note that this is not a blocking comment, just something I want to have a discussion about.

Just to make it clear: this will update all direct and indirect dependencies latest minor or patch releases. If there are now releases, this will update to the latest commit on the main branch.

I think the term `update` hides what is happening there. `go get -u` will increases the **required** version of the modules. Requiring implies that you have a reason to require that newest version. But in the case of `go get -u` you usually don't have, particularly with so many dependencies like this project.


After reading https://research.swtch.com/vgo-mvs I think that dependencies updates should be "Predictable. Boring.". vgo/go modules follows a different pattern then the usual "always use the latest version" pattern we know from e.g. npm.


Please also note this comment from https://github.com/golang/go/issues/24301#issuecomment-371228664:

> Given that minimal version selection takes the minimum allowed version of each dependency, it's easy to think that this would lead to use of very old copies of packages, which in turn might lead to unnecessary bugs or security problems. In practice, however, I think the opposite will happen, because the minimum allowed version is the maximum of all the constraints, so the one lever of control made available to all modules in a build is the ability to force the use of a newer version of a dependency than would otherwise be used. I expect that users of minimal version selection will end up with programs that are almost as up-to-date as their friends using more aggressive systems like Cargo.
> For example, suppose you are writing a program that depends on a handful of other modules, all of which depend on some very common module, like gopkg.in/yaml.v2. Your program's build will use the newest YAML version among the ones requested by your module and that handful of dependencies. Even just one conscientious dependency can force your build to update many other dependencies. This is the opposite of the Kubernetes Go client problem I mentioned earlier.
> If anything, minimal version selection would instead suffer the opposite problem, that this “max of the minimums” answer serves as a ratchet that forces dependencies forward too quickly. But I think in practice dependencies will move forward at just the right speed, which ends up being just the right amount slower than Cargo and friends.

Go is about stability. And using the "bleeding edge" of every decencies just don't seam aligned with this idea.

Reply:
I get minimum version selection. What this does is move all the minimum versions forward. This pulls in the latest bug/security fixes.

> Requiring implies that you have a reason to require that newest version.

We do. We want to use the latest versions of our dependencies.

> Go is about stability. And using the "bleeding edge" of every decencies just don't seam aligned with this idea.

I am not sure what you are asking us to do here. Never update our dependencies?

Reply:
What do you think about using `go get -u=patch` instead? This ensure we always get the latest bug fixes.